File: src/main/java/net/katsstuff/danmakucore/entity/spellcard/SpellcardDummy.java
/*
 * This class was created by <Katrix>. It's distributed as
 * part of the DanmakuCore Mod. Get the Source Code in github:
 * https://github.com/Katrix-/DanmakuCore
 *
 * DanmakuCore is Open Source and distributed under the
 * the DanmakuCore license: https://github.com/Katrix-/DanmakuCore/blob/master/LICENSE.md
 */
package net.katsstuff.danmakucore.entity.spellcard;

import javax.annotation.Nullable;

import net.katsstuff.danmakucore.entity.living.boss.EnumTouhouCharacters;
import net.minecraft.entity.EntityLivingBase;

public class SpellcardDummy extends Spellcard {

	@Override
	public SpellcardEntity instantiate(EntitySpellcard card, @Nullable EntityLivingBase target) {
		return new SpellcardEntity(this, card, target) {

			@Override
			public void onSpellcardUpdate() {}
		};
	}

	@Override
	public int getNeededLevel() {
		return 0;
	}

	@Override
	public int getRemoveTime() {
		return 0;
	}

	@Override
	public int getEndTime() {
		return 0;
	}

	@Override
	public EnumTouhouCharacters getOriginalUser() {
		return EnumTouhouCharacters.OTHER;
	}
}

File: src/main/scala/net/katsstuff/teamnightclipse/danmakucore/entity/spellcard/SpellcardDummy.scala
/*
 * Copyright (C) 2018  Katrix
 * This file is part of DanmakuCore.
 *
 * DanmakuCore is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * DanmakuCore is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with DanmakuCore.  If not, see <https://www.gnu.org/licenses/>.
 */
package net.katsstuff.teamnightclipse.danmakucore.entity.spellcard

import net.minecraft.entity.EntityLivingBase

object SpellcardDummy extends Spellcard {

  def instance: SpellcardDummy.type = this

  override def instantiate(card: EntitySpellcard, target: Option[EntityLivingBase]): SpellcardEntity =
    new SpellcardEntity(this, card, target) {
      override def onSpellcardUpdate(): Unit = {}
    }

  override def level = 0

  override def removeTime = 0

  override def endTime = 0

  override def touhouUser = throw new IllegalStateException("Tried to get user of dummy spellcard")
}
